### PR TITLE
CLN: Assorted cleanups

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/tests/io/test_html.py [hunk 1]
```diff
@@ -262,6 +262,7 @@ def test_bad_url_protocol(self):
             self.read_html('git://github.com', match='.*Water.*')
 
     @network
+    @pytest.mark.slow
     def test_invalid_url(self):
```
Comment (jbrockmendel) [inline pandas/tests/io/test_html.py]: two versions of this test get run and each take 150s locally

File: pandas/core/reshape/merge.py [hunk 2]
```diff
@@ -1809,3 +1809,44 @@ def validate_operand(obj):
     else:
         raise TypeError('Can only merge Series or DataFrame objects, '
                         'a {obj} was passed'.format(obj=type(obj)))
+
+
+def _items_overlap_with_suffix(left, lsuffix, right, rsuffix):
```
Comment (jreback) [inline pandas/core/reshape/merge.py]: can be a later PR, can you type these args

File: pandas/core/reshape/merge.py [hunk 3]
```diff
@@ -1809,3 +1809,44 @@ def validate_operand(obj):
...
+    """
+    If two indices overlap, add suffixes to overlapping entries.
+
+    If corresponding suffix is empty, the entry is simply converted to string.
+
+    """
+    to_rename = left.intersection(right)
+    if len(to_rename) == 0:
+        return left, right
+    else:
```
Comment (jreback) [inline pandas/core/reshape/merge.py]: don't need the else here (just dedent)

**Review by jreback — APPROVED** [review]
ok with merging or comments as you want @jbrockmendel

**Comment by jreback** [comment]
thanks

**Pull request merged**